client/src/app/pages/issues/helpers.ts

```diff
@@ -62,16 +63,28 @@ export const useSharedAffectedApplicationFilterCategories = <
         t("actions.filterBy", {
           what: t("terms.archetype").toLowerCase(),
         }) + "...",
-      selectOptions: archetypes.map(({ name }) => ({ key: name, value: name })),
-      getServerFilterValue: (selectedOptions) =>
-        selectedOptions
-          ?.map((option) => archetypes.find((item) => item.name === option))
+      selectOptions: archetypes
+        .filter(({ applications }) => applications && applications.length > 0)
```

Comment:
This still filters out archetypes w/o applications.  The hack in `getServerFilterValue` allows selection of an "empty" archetype.